File: leetcode/leetcode_p226/code_226.py
# 力扣第226题，翻转二叉树
"""
翻转一棵二叉树
"""


class TreeNode:
    def __init__(self, x) -> None:
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def invertTree(self, root):
        if not root:
            return []
        
        def dfs(node):
            if not node:
                return
            node.left, node.right = node.right, node.left
            dfs(node.left)
            dfs(node.right)

File: leetcode/leetcode_p226/code_226.cpp
// 力扣第226题，翻转二叉树
/*
翻转一棵二叉树
*/
#include <iostream>
using namespace std;

struct TreeNode {
    int val;
    TreeNode * left;
    TreeNode * right;
    TreeNode(int x): val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
    TreeNode* invertTree(TreeNode* root){
        if (root == nullptr)
            return root;
        
        TreeNode* l = invertTree(root->left);
        TreeNode* r = invertTree(root->right);
        root->left = r;
        root->right = l;
        return root;
    }
};

